extra_data/write_cxi.py

```diff
@@ -20,16 +23,14 @@ class VirtualCXIWriter:
 
     Parameters
     ----------
-
-    detdata: extra_data.components.MPxDetectorBase
+    detdata: extra_data.components.MultimodDetectorBase
       The detector data interface for the data to gather in this file.
     """
     def __init__(self, detdata):
         self.detdata = detdata
+        self.group_label, self.image_label = detdata._main_data_key.split('.')
 
-        self.modulenos = sorted(detdata.modno_to_source)
-
-        frame_counts = detdata.frame_counts
+        frame_counts = detdata.frame_counts * self.cells_per_entry
```

Comment:
Pycharm gives me quite a few warnings because attributes like `.cells_per_entry` are used in the base class but only defined in the child classes. I might give them some defaults in the base class to help tools like this follow the code (it can also help human readers).